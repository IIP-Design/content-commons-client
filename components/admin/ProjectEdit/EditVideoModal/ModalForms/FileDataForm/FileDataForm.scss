@import "styles/colors.scss";

.file_meta_content {
  color: $blue;
  display: block;
  font-size: 0.8rem;

  &--filetype {
    margin-bottom: 0.5em;
    // for long strings in title - break word w/ hyphens if browser support, otherwise fallback to wrap
    overflow-wrap: break-word;
    word-wrap: break-word;
    -webkit-hyphens: auto;
    -ms-hyphens: auto;
    -moz-hyphens: auto;
    hyphens: auto;
  }
}

.video-file-form {
  margin: 1rem;

  &-col-1 {
    display: flex !important;
    flex-direction: column;
    justify-content: space-between;
  }
}

.video-links {
  @media screen and (max-width: 991px) {
    margin: 1rem 0;
  }
}

.delete-file-link {
  color: $light-blue;
  cursor: pointer;
  display: block;
  font-size: 0.8rem;
  margin-top: 0.3rem;
}
