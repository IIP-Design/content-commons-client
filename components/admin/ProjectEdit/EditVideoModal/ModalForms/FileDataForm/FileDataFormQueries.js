import { gql } from '@apollo/client';

export const VIDEO_PROJECT_QUERY = gql`
  query VIDEO_PROJECT_QUERY( $id: ID! ) {
    project: videoProject( id: $id ) {
      id
      projectTitle
      units {
        id
        language {
          id
          locale
        }
      }
    }
  }
`;

export const VIDEO_FILE_QUERY = gql`
  query VIDEO_FILE_QUERY( $id: ID! ) {
    file: videoFile( id: $id ) {
      id
      createdAt
      duration
      filename
      filesize
      quality
      videoBurnedInStatus
      dimensions {
        id
        height
        width
      }
      language {
        id
        displayName
      }
      stream {
        id
        site
        url
      }
      use {
        id
        name
      }
    }
  }
`;

export const VIDEO_FILE_LANG_MUTATION = gql`
  mutation VIDEO_FILE_LANG_MUTATION( $id: ID!, $language: ID! ) {
    updateVideoFile(
      data: {
        language: {
          connect: { id: $language }
        }
      },
      where: { id: $id }
    ) {
      id
      language { id }
    }
  }
`;

export const VIDEO_UNIT_CONNECT_FILE_MUTATION = gql`
  mutation VIDEO_UNIT_CONNECT_FILE_MUTATION( $id: ID!, $fileId: ID! ) {
    updateVideoUnit(
      data: {
        files: {
          connect: { id: $fileId }
        }
      },
      where: { id: $id }
    ) {
      id
      files { id }
    }
  }
`;

export const VIDEO_UNIT_DISCONNECT_FILE_MUTATION = gql`
  mutation VIDEO_UNIT_DISCONNECT_FILE_MUTATION( $id: ID!, $fileId: ID! ) {
    updateVideoUnit(
      data: {
        files: {
          disconnect: { id: $fileId }
        }
      },
      where: { id: $id }
    ) {
      id
      files { id }
    }
  }
`;

export const VIDEO_FILE_SUBTITLES_MUTATION = gql`
  mutation VIDEO_FILE_SUBTITLES_MUTATION( $id: ID!, $videoBurnedInStatus: VideoBurnedInStatus! ) {
    updateVideoFile(
      data: { videoBurnedInStatus: $videoBurnedInStatus },
      where: { id: $id }
    ) {
      id
      videoBurnedInStatus
    }
  }
`;

export const VIDEO_FILE_USE_MUTATION = gql`
  mutation VIDEO_FILE_USE_MUTATION( $id: ID!, $use: ID! ) {
    updateVideoFile(
      data: {
        use: {
          connect: { id: $use }
        }
      },
      where: { id: $id }
    ) {
      id
      use { id }
    }
  }
`;

export const VIDEO_FILE_QUALITY_MUTATION = gql`
  mutation VIDEO_FILE_QUALITY_MUTATION( $id: ID!, $quality: VideoQuality! ) {
    updateVideoFile(
      data: { quality: $quality },
      where: { id: $id }
    ) {
      id
      quality
    }
  }
`;

export const VIDEO_FILE_CREATE_STREAM_MUTATION = gql`
  mutation VIDEO_FILE_CREATE_STREAM_MUTATION( $id: ID!, $site: String!, $url: String! ) {
    updateVideoFile(
      data: { 
        stream: {
          create: {
            site: $site,
            url: $url
          }
        }
      },
      where: { id: $id }
    ) {
      id
      stream { id }
    }
  }
`;

export const VIDEO_FILE_UPDATE_STREAM_MUTATION = gql`
  mutation VIDEO_FILE_UPDATE_STREAM_MUTATION( $id: ID!, $streamId: ID! $url: String! ) {
    updateVideoFile(
      data: {
        stream: {
          update: {
            data: {
              url: $url
            },
            where: {
              id: $streamId
            }
          }
        }
      },
      where: { id: $id }
    ) {
      id
      stream { id }
    }
  }
`;

export const VIDEO_FILE_DELETE_STREAM_MUTATION = gql`
  mutation VIDEO_FILE_DELETE_STREAM_MUTATION( $id: ID!, $streamId: ID! ) {
    updateVideoFile(
      data: {
        stream: {
          delete: { id: $streamId }
        }
      },
      where: { id: $id }
    ) {
      id
      stream { id }
    }
  }
`;

export const VIDEO_FILE_DELETE_MUTATION = gql`
  mutation VIDEO_FILE_DELETE_MUTATION( $id: ID! ) {
    deleteVideoFile( id: $id ) {
      id
    }
  }
`;
