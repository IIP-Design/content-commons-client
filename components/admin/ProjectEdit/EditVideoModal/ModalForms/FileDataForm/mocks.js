import * as query from './FileDataFormQueries';
import { LANGUAGES_QUERY, LANGUAGE_BY_NAME_QUERY } from 'components/admin/dropdowns/LanguageDropdown/LanguageDropdown';
import { VIDEO_UNIT_QUERY } from 'components/admin/ProjectEdit/EditVideoModal/ModalSections/FileSection/FileSection';
import { VIDEO_USE_QUERY } from 'components/admin/dropdowns/UseDropdown/UseDropdown';

const props = {
  fileCount: 2,
  setFieldValue: jest.fn(),
  values: {
    language: 'ck2lzfx710hkq07206thus6pt',
    quality: 'WEB',
    use: 'ckcm6aii005dv0720fapgz92p',
    videoBurnedInStatus: 'CLEAN',
    vimeo: 'https://vimeo.com/438865910',
    youtube: '',
  },
};

const reduxState = {
  projectUpdated: false,
};

const mocks = [
  {
    request: {
      query: query.VIDEO_PROJECT_QUERY,
      variables: {
        id: 'ckcoositc0biv0720dwlj4l0g',
      },
    },
    result: {
      data: {
        project: {
          id: 'ckcoositc0biv0720dwlj4l0g',
          units: [
            {
              id: 'ckcoospdp0bjc07207x6j8xhw',
              language: {
                id: 'ck2lzfx710hkq07206thus6pt',
                locale: 'en-us',
                __typename: 'Language',
              },
              __typename: 'VideoUnit',
            },
            {
              id: 'ckcovuyt30co70720tp5w6oem',
              language: {
                id: 'ck2lzfx710hkp07206oo0icbv',
                locale: 'fr-fr',
                __typename: 'Language',
              },
              __typename: 'VideoUnit',
            },
          ],
          __typename: 'VideoProject',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_QUERY,
      variables: {
        // props.selectedFile from redux
        id: 'ckcoospf10bjh0720ftniacm3',
      },
    },
    result: {
      data: {
        file: {
          id: 'ckcoospf10bjh0720ftniacm3',
          createdAt: '2020-07-16T11:05:08.192Z',
          duration: 14.889875,
          filename: 'Mexico City 1.mp4',
          filesize: 15829673,
          quality: 'WEB',
          videoBurnedInStatus: 'CLEAN',
          dimensions: {
            id: 'ckcoosph00bjm07205nrbnxpu',
            height: 1080,
            width: 1920,
            __typename: 'Dimensions',
          },
          language: {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            __typename: 'Language',
          },
          stream: [
            {
              id: 'ckcoospi00bjn0720nt6p5ful',
              site: 'vimeo',
              url: 'https://vimeo.com/438865910',
              __typename: 'VideoStream',
            },
          ],
          use: {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
            __typename: 'VideoUse',
          },
          __typename: 'VideoFile',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_LANG_MUTATION,
      variables: {
        data: {
          language: {
            connect: { id: 'ck2lzfx710hkp07206oo0icbv' },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          language: { id: 'ck2lzfx710hkp07206oo0icbv' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_UNIT_CONNECT_FILE_MUTATION,
      variables: {
        data: {
          files: {
            connect: { id: 'ckcoospf10bjh0720ftniacm3' },
          },
        },
        where: { id: 'ckcoospdp0bjc07207x6j8xhw' },
      },
    },
    result: {
      data: {
        updateVideoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: { id: 'ckcoospf10bjh0720ftniacm3' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_UNIT_DISCONNECT_FILE_MUTATION,
      variables: {
        data: {
          files: {
            disconnect: { id: 'ckcoospf10bjh0720ftniacm3' },
          },
        },
        where: { id: 'ckcoospdp0bjc07207x6j8xhw' },
      },
    },
    result: {
      data: {
        updateVideoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: { id: 'ckcoospf10bjh0720ftniacm3' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_SUBTITLES_MUTATION,
      variables: {
        data: {
          videoBurnedInStatus: 'SUBTITLED',
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          videoBurnedInStatus: 'SUBTITLED',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_USE_MUTATION,
      variables: {
        data: {
          use: {
            connect: { id: 'ck2lzfx5y0hhz07207nol9j6r' },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          use: { id: 'ck2lzfx5y0hhz07207nol9j6r' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_QUALITY_MUTATION,
      variables: {
        data: {
          quality: 'WEB',
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          quality: 'WEB',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_CREATE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            create: {
              site: 'thesite',
              url: 'https://thesite.com/the-site-stream-id',
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_UPDATE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            update: {
              data: {
                url: 'https://thesite.com/the-new-site-stream-id',
              },
              where: {
                id: 'the-new-site-stream-id',
              },
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-new-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_DELETE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            'delete': {
              id: 'the-site-stream-id',
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_DELETE_MUTATION,
      variables: {
        id: 'ckcoospf10bjh0720ftniacm3',
      },
    },
    result: {
      data: {
        deleteVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
        },
      },
    },
  },
  {
    request: {
      query: LANGUAGE_BY_NAME_QUERY,
      variables: {
        where: { displayName: 'English' },
      },
    },
    result: {
      data: {
        languages: [
          {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_USE_QUERY,
      variables: {
        where: { name: 'Clean' },
      },
    },
    result: {
      data: {
        videoUses: [
          {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_UNIT_QUERY,
      variables: { id: 'ckcoospdp0bjc07207x6j8xhw' },
    },
    result: {
      data: {
        videoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: [
            {
              id: 'ckcoospf10bjh0720ftniacm3',
            },
          ],
          language: {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        },
      },
    },
  },
  {
    request: {
      query: LANGUAGES_QUERY,
    },
    result: {
      data: {
        languages: [
          {
            id: 'ck2lzfx710hkp07206oo0icbv',
            displayName: 'French',
            locale: 'fr-fr',
          },
          {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_USE_QUERY,
    },
    result: {
      data: {
        videoUses: [
          {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
          },
          {
            id: 'ck2lzfx5y0hhz07207nol9j6r',
            name: 'Full Video',
          },
        ],
      },
    },
  },
];

export { mocks, props, reduxState };
