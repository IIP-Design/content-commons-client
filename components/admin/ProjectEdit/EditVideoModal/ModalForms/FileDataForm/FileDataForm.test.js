import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from '@apollo/react-testing';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { mocks, props, reduxState } from './mocks';
import { suppressActWarning } from 'lib/utils';
import { EditSingleProjectItemContext } from 'components/admin/ProjectEdit/EditSingleProjectItem/EditSingleProjectItem';
import FileDataForm from './FileDataForm';

jest.mock(
  'next/dynamic',
  () => function DynamicComponent() { return ''; },
);
jest.mock(
  'components/admin/ConfirmModalContent/ConfirmModalContent',
  () => function ConfirmModalContent() { return ''; },
);
jest.mock(
  'components/admin/dropdowns/QualityDropdown/QualityDropdown',
  () => function QualityDropdown() { return ''; },
);
jest.mock(
  'components/admin/dropdowns/VideoBurnedInStatusDropdown/VideoBurnedInStatusDropdown',
  () => function VideoBurnedInStatusDropdown() { return ''; },
);

const value = {
  language: {
    id: 'ck2lzfx710hkq07206thus6pt',
    displayName: 'English',
    locale: 'en-us',
    __typename: 'Language',
  },
  selectedFile: 'ckcoospf10bjh0720ftniacm3',
  selectedProject: 'ckcoositc0biv0720dwlj4l0g',
  selectedUnit: 'ckcoospdp0bjc07207x6j8xhw',
  setLanguage: jest.fn(),
  setSelectedFile: jest.fn(),
  setSelectedUnit: jest.fn(),
  setSelectedProject: jest.fn(),
  setShowNotification: jest.fn(),
  showNotication: jest.fn(),
  startTimeout: jest.fn(),
  updateSelectedUnit: jest.fn(),
};

const mockStore = configureStore( [] );

describe( '<FileDataForm />, when the video use is Clean', () => {
  const consoleError = console.error;

  beforeAll( () => suppressActWarning( consoleError ) );
  afterAll( () => {
    console.error = consoleError;
  } );

  let store;
  let Component;
  let wrapper;

  beforeEach( () => {
    store = mockStore( reduxState );
    Component = (
      <EditSingleProjectItemContext.Provider value={ value }>
        <Provider store={ store }>
          <MockedProvider mocks={ mocks } addTypename={ false }>
            <FileDataForm { ...{ ...props, ...value } } />
          </MockedProvider>
        </Provider>
      </EditSingleProjectItemContext.Provider>
    );
    wrapper = mount( Component );
  } );

  it( 'renders initial loading state without crashing', () => {
    const fileDataForm = wrapper.find( 'FileDataForm' );
    const loader = wrapper.find( 'Loader' );
    const msg = 'Loading the file data...';

    expect( fileDataForm.exists() ).toEqual( true );
    expect( loader.exists() ).toEqual( true );
    expect( loader.contains( msg ) ).toEqual( true );
  } );

  it( 'renders final state without crashing', async () => {
    await wait( 0 );
    wrapper.update();
    const form = wrapper.find( 'Form' );
    const loader = wrapper.find( 'Loader' );

    expect( form.exists() ).toEqual( true );
    expect( form.hasClass( 'edit-video__form video-file-form' ) )
      .toEqual( true );
    expect( loader.exists() ).toEqual( false );
  } );

  it( 'renders file meta data', async () => {
    await wait( 0 );
    wrapper.update();
    const fileMeta = wrapper.find( '.file_meta' );

    expect( toJSON( fileMeta ) ).toMatchSnapshot();
  } );

  it( 'renders the Confirm modal', async () => {
    await wait( 0 );
    wrapper.update();
    const confirm = wrapper.find( 'Confirm' );

    expect( toJSON( confirm ) ).toMatchSnapshot();
  } );

  it( 'renders video links', async () => {
    await wait( 0 );
    wrapper.update();
    const videoLinks = wrapper.find( '.video-links' );

    expect( toJSON( videoLinks ) ).toMatchSnapshot();
  } );

  it( 'renders the LanguageDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const languageDropdown = wrapper.find( 'LanguageDropdown' );
    const languageId = mocks[1].result.data.file.language.id;

    expect( languageDropdown.exists() ).toEqual( true );
    expect( languageDropdown.prop( 'label' ) ).toEqual( 'Language' );
    expect( languageDropdown.prop( 'required' ) ).toEqual( true );
    expect( languageDropdown.prop( 'value' ) ).toEqual( languageId );
    // disabled since file (i.e., mocks[1]) has Clean video use
    expect( languageDropdown.prop( 'disabled' ) ).toEqual( true );
  } );

  it( 'renders the VideoBurnedInStatusDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const burnedInStatusDropdown = wrapper.find( 'VideoBurnedInStatusDropdown' );

    expect( burnedInStatusDropdown.prop( 'label' ) )
      .toEqual( 'On-Screen Text' );
    expect( burnedInStatusDropdown.prop( 'name' ) )
      .toEqual( 'videoBurnedInStatus' );
    expect( burnedInStatusDropdown.prop( 'type' ) )
      .toEqual( 'video' );
    expect( burnedInStatusDropdown.prop( 'required' ) )
      .toEqual( true );
    // disabled since file (i.e., mocks[1]) has Clean video use
    expect( burnedInStatusDropdown.prop( 'disabled' ) )
      .toEqual( true );
    expect( burnedInStatusDropdown.prop( 'onChange' ).name )
      .toEqual( 'handleDropdownSave' );
  } );

  it( 'renders the UseDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const useDropdown = wrapper.find( 'UseDropdown' );
    const useId = mocks[1].result.data.file.use.id;

    expect( useDropdown.prop( 'label' ) ).toEqual( 'Video Type' );
    expect( useDropdown.prop( 'name' ) ).toEqual( 'use' );
    expect( useDropdown.prop( 'type' ) ).toEqual( 'video' );
    expect( useDropdown.prop( 'required' ) ).toEqual( true );
    expect( useDropdown.prop( 'value' ) ).toEqual( useId );
    expect( useDropdown.prop( 'onChange' ).name )
      .toEqual( 'handleDropdownSave' );
  } );

  it( 'renders the QualityDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const qualityDropdown = wrapper.find( 'QualityDropdown' );
    const infoTip = 'Web: small - for social sharing, Broadcast: large - ambassador videos';

    expect( qualityDropdown.prop( 'label' ) )
      .toEqual( 'Video Quality' );
    expect( qualityDropdown.prop( 'name' ) ).toEqual( 'quality' );
    expect( qualityDropdown.prop( 'type' ) ).toEqual( 'video' );
    expect( qualityDropdown.prop( 'infotip' ) ).toEqual( infoTip );
    expect( qualityDropdown.prop( 'required' ) ).toEqual( true );
    expect( qualityDropdown.prop( 'onChange' ).name )
      .toEqual( 'handleDropdownSave' );
  } );
} );

describe( '<FileDataForm />, when the video use is not Clean', () => {
  const consoleError = console.error;

  beforeAll( () => suppressActWarning( consoleError ) );
  afterAll( () => {
    console.error = consoleError;
  } );

  let store;
  let Component;
  let wrapper;

  const french = {
    id: 'ck2lzfx710hkp07206oo0icbv',
    displayName: 'French',
    locale: 'fr-fr',
    __typename: 'Language',
  };

  const newCtxValue = {
    ...value,
    language: french,
  };

  const newProps = {
    ...props,
    values: {
      ...props.values,
      language: french.id,
      use: 'ck2lzfx5y0hhz07207nol9j6r',
      videoBurnedInStatus: 'SUBTITLED',
    },
  };

  const newMocks = mocks.reduce( ( acc, curr, i ) => {
    if ( i === 1 ) {
      acc.push( {
        ...curr,
        result: {
          data: {
            file: {
              ...curr.result.data.file,
              language: french,
              use: {
                id: 'ck2lzfx5y0hhz07207nol9j6r',
                name: 'Full Video',
              },
            },
          },
        },
      } );
    } else {
      acc.push( curr );
    }

    return acc;
  }, [] );

  beforeEach( () => {
    store = mockStore( reduxState );
    Component = (
      <EditSingleProjectItemContext.Provider value={ newCtxValue }>
        <Provider store={ store }>
          <MockedProvider mocks={ newMocks } addTypename={ false }>
            <FileDataForm { ...{ ...newProps, ...newCtxValue } } />
          </MockedProvider>
        </Provider>
      </EditSingleProjectItemContext.Provider>
    );
    wrapper = mount( Component );
  } );

  it( 'renders initial loading state without crashing', () => {
    const fileDataForm = wrapper.find( 'FileDataForm' );
    const loader = wrapper.find( 'Loader' );
    const msg = 'Loading the file data...';

    expect( fileDataForm.exists() ).toEqual( true );
    expect( loader.exists() ).toEqual( true );
    expect( loader.contains( msg ) ).toEqual( true );
  } );

  it( 'renders final state without crashing', async () => {
    await wait( 0 );
    wrapper.update();
    const form = wrapper.find( 'Form' );
    const loader = wrapper.find( 'Loader' );

    expect( form.exists() ).toEqual( true );
    expect( form.hasClass( 'edit-video__form video-file-form' ) )
      .toEqual( true );
    expect( loader.exists() ).toEqual( false );
  } );

  it( 'renders the LanguageDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const languageDropdown = wrapper.find( 'LanguageDropdown' );
    const languageId = newMocks[1].result.data.file.language.id;

    expect( languageDropdown.exists() ).toEqual( true );
    expect( languageDropdown.prop( 'label' ) ).toEqual( 'Language' );
    expect( languageDropdown.prop( 'required' ) ).toEqual( true );
    expect( languageDropdown.prop( 'value' ) ).toEqual( languageId );
    // disabled since file (i.e., mocks[1]) has Clean video use
    expect( languageDropdown.prop( 'disabled' ) ).toEqual( false );
  } );

  it( 'renders the VideoBurnedInStatusDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const burnedInStatusDropdown = wrapper.find( 'VideoBurnedInStatusDropdown' );

    expect( burnedInStatusDropdown.prop( 'label' ) )
      .toEqual( 'On-Screen Text' );
    expect( burnedInStatusDropdown.prop( 'name' ) )
      .toEqual( 'videoBurnedInStatus' );
    expect( burnedInStatusDropdown.prop( 'type' ) )
      .toEqual( 'video' );
    expect( burnedInStatusDropdown.prop( 'required' ) )
      .toEqual( true );
    // disabled since file (i.e., mocks[1]) has Clean video use
    expect( burnedInStatusDropdown.prop( 'disabled' ) )
      .toEqual( false );
    expect( burnedInStatusDropdown.prop( 'onChange' ).name )
      .toEqual( 'handleDropdownSave' );
  } );

  it( 'renders the UseDropdown', async () => {
    await wait( 0 );
    wrapper.update();
    const useDropdown = wrapper.find( 'UseDropdown' );
    const useId = newMocks[1].result.data.file.use.id;

    expect( useDropdown.prop( 'label' ) ).toEqual( 'Video Type' );
    expect( useDropdown.prop( 'name' ) ).toEqual( 'use' );
    expect( useDropdown.prop( 'type' ) ).toEqual( 'video' );
    expect( useDropdown.prop( 'required' ) ).toEqual( true );
    expect( useDropdown.prop( 'value' ) ).toEqual( useId );
    expect( useDropdown.prop( 'onChange' ).name )
      .toEqual( 'handleDropdownSave' );
  } );
} );

describe( '<FileDataForm />, when there is a GraphQL error for the videoFileQuery', () => {
  const consoleError = console.error;

  beforeAll( () => suppressActWarning( consoleError ) );
  afterAll( () => {
    console.error = consoleError;
  } );

  let store;
  let Component;
  let wrapper;

  const newMocks = mocks.reduce( ( acc, curr, i ) => {
    if ( i === 1 ) {
      acc.push( {
        ...curr,
        result: {
          errors: [{ message: 'There was an error.' }],
        },
      } );
    } else {
      acc.push( curr );
    }

    return acc;
  }, [] );

  beforeEach( () => {
    store = mockStore( reduxState );
    Component = (
      <EditSingleProjectItemContext.Provider value={ value }>
        <Provider store={ store }>
          <MockedProvider mocks={ newMocks } addTypename={ false }>
            <FileDataForm { ...{ ...props, ...value } } />
          </MockedProvider>
        </Provider>
      </EditSingleProjectItemContext.Provider>
    );
    wrapper = mount( Component );
  } );

  it( 'renders initial loading state without crashing', () => {
    const fileDataForm = wrapper.find( 'FileDataForm' );
    const loader = wrapper.find( 'Loader' );
    const msg = 'Loading the file data...';

    expect( fileDataForm.exists() ).toEqual( true );
    expect( loader.exists() ).toEqual( true );
    expect( loader.contains( msg ) ).toEqual( true );
  } );

  it( 'renders the GraphQL error message', async () => {
    await wait( 0 );
    wrapper.update();
    const form = wrapper.find( 'Form' );
    const apolloError = wrapper.find( 'ApolloError' );
    const msg = 'There was an error.';

    expect( form.exists() ).toEqual( false );
    expect( apolloError.exists() ).toEqual( true );
    expect( apolloError.contains( msg ) ).toEqual( true );
  } );
} );
