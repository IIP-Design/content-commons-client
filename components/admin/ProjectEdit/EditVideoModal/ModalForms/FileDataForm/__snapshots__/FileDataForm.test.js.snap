// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileDataForm />, when the video use is Clean renders file meta data 1`] = `
<div
  className="file_meta"
>
  <span
    className="file_meta_content file_meta_content--filetype"
    lang="en"
  >
    Mexico City 1.mp4
  </span>
  <span
    className="file_meta_content file_meta_content--filesize"
  >
    Filesize: 15.1  MB
  </span>
  <span
    className="file_meta_content file_meta_content--dimensions"
  >
    Dimensions: 1920 x 1080
  </span>
  <span
    className="file_meta_content file_meta_content--uploaded"
  >
    Uploaded: July 16, 2020
  </span>
  <span
    className="file_meta_content file_meta_content--duration"
  >
    Duration: 0:14
  </span>
  <span
    className="delete-file-link"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    Delete file from project
  </span>
</div>
`;

exports[`<FileDataForm />, when the video use is Clean renders the Confirm modal 1`] = `
<Confirm
  cancelButton="No, take me back"
  className="delete"
  confirmButton="Yes, delete forever"
  content={
    <ConfirmModalContent
      className="delete_confirm delete_confirm--video"
      headline="Are you sure you want to deleteÂ this video?"
    >
      <p>
        This video will be permanently removed from the Content Commons and any other projects or collections it appears on.
      </p>
    </ConfirmModalContent>
  }
  onCancel={[Function]}
  onConfirm={[Function]}
  open={false}
  size="small"
>
  <Modal
    centered={true}
    className="delete"
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="small"
  >
    <Portal
      closeOnDocumentClick={false}
      closeOnEscape={true}
      eventPool="Modal"
      mountNode={<body />}
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      open={false}
      openOnTriggerClick={true}
    />
  </Modal>
</Confirm>
`;

exports[`<FileDataForm />, when the video use is Clean renders video links 1`] = `
<div
  className="video-links"
>
  <FormInput
    as={[Function]}
    control={[Function]}
    id="video-youtube"
    label="YouTube URL"
    name="youtube"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  >
    <FormField
      control={[Function]}
      id="video-youtube"
      label="YouTube URL"
      name="youtube"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    >
      <div
        className="field"
      >
        <label
          htmlFor="video-youtube"
        >
          YouTube URL
        </label>
        <Input
          id="video-youtube"
          name="youtube"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        >
          <div
            className="ui input"
          >
            <input
              id="video-youtube"
              name="youtube"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </Input>
      </div>
    </FormField>
  </FormInput>
  <FormInput
    as={[Function]}
    control={[Function]}
    id="video-vimeo"
    label="Vimeo URL"
    name="vimeo"
    onBlur={[Function]}
    onChange={[Function]}
    readOnly={true}
    style={
      Object {
        "opacity": "0.45",
      }
    }
    value="https://vimeo.com/438865910"
  >
    <FormField
      control={[Function]}
      id="video-vimeo"
      label="Vimeo URL"
      name="vimeo"
      onBlur={[Function]}
      onChange={[Function]}
      readOnly={true}
      style={
        Object {
          "opacity": "0.45",
        }
      }
      value="https://vimeo.com/438865910"
    >
      <div
        className="field"
      >
        <label
          htmlFor="video-vimeo"
        >
          Vimeo URL
        </label>
        <Input
          id="video-vimeo"
          name="vimeo"
          onBlur={[Function]}
          onChange={[Function]}
          readOnly={true}
          style={
            Object {
              "opacity": "0.45",
            }
          }
          type="text"
          value="https://vimeo.com/438865910"
        >
          <div
            className="ui input"
            style={
              Object {
                "opacity": "0.45",
              }
            }
          >
            <input
              id="video-vimeo"
              name="vimeo"
              onBlur={[Function]}
              onChange={[Function]}
              readOnly={true}
              type="text"
              value="https://vimeo.com/438865910"
            />
          </div>
        </Input>
      </div>
    </FormField>
  </FormInput>
</div>
`;
